pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/ConnectionPool.java

```diff
@@ -170,6 +189,15 @@ public void initChannel(SocketChannel ch) throws Exception {
                 cnx.ctx().close();
                 return null;
             });
+
+            // close the connection after connection-lifetime completes
+            if (connectionLifetimeInSecond > 0) {
+                scheduler.schedule(() -> {
+                    if (cnx != null && cnx.channel().isActive()) {
+                        cnx.channel().disconnect();
+                    }
+                }, connectionLifetimeInSecond, TimeUnit.SECONDS);
+            }
```

Comment:
I think our goal is to not keep lookup connection persistent forever and close after n minute. however, we can add check to close connection if not active for last n minute. i will add another commit.